projects/plugins/jetpack/class.jetpack.php

```diff
@@ -4443,31 +4443,22 @@ public function build_connect_url( $raw = false, $redirect = false, $from = fals
 	 * Create the Jetpack authorization URL.
 	 *
 	 * @param bool|string $redirect URL to redirect to.
-	 * @param bool        $iframe Whether to use the iframe version.
 	 *
 	 * @todo Update default value for redirect since the called function expects a string.
 	 *
 	 * @return mixed|void
 	 */
-	public static function build_authorize_url( $redirect = false, $iframe = false ) {
+	public static function build_authorize_url( $redirect = false ) {
```

Comment:
In theory no, but we could just deprecate the argument and leave it there for a while...

Reply:
In theory no, but we could just deprecate the argument and leave it there for a while...

Reply:
I think we should keep the old signature, and delete it a few versions later. It would cost us nothing, and might save a bunch of headache to others (possibly others too).
It's easy to miss something, and there may be other plugins using it the old way.

We could rename the second parameter to `$deprecated`, and throw the `_doing_it_wrong()` notice if's passed.

Here's an example of the missed call with two arguments: https://github.com/Automattic/jetpack/blob/update/remove-remaining-in-place-conntections/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#L1704
The method is deprecated, but might be used nonetheless.

Reply:
Agreed, let's add the original signature back to this method. 

removing register_site() sounds a bit scary 😅 . Let's err on the side of caution for this round.